staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/context.go

```diff
@@ -0,0 +1,57 @@
+package validation
+
+import (
+	"context"
+
+	"k8s.io/apiextensions-apiserver/pkg/apiserver/validation/metrics"
+	"k8s.io/apiserver/pkg/cel/common"
+)
+
+type sharedRatchetingContextInfo struct {
+	// Cached correlation and CumulativeMeasurement information between the old and new objects
+	// for performance. Initially nil. Populated upon first call
+	correlation *common.CorrelatedObject
+}
+
+type sharedRatchetingContextKeyType string
+
+const sharedRatchetingContextKey sharedRatchetingContextKeyType = "sharedRatchetingContextKey"
+
+// WithCRDValidationRatcheting returns a new context with a CorrelatedObject
+// associated it for the validation of a custom resource.
+//
+// This function should be called at the beginning of a request
+// PublishRatchetingMetrics should be called at the end of a request
+func WithCRDValidationRatcheting(ctx context.Context, new, old interface{}, schema common.Schema) context.Context {
```

Comment:
Really prefer not to use `ctx` to pass the `CorrelatedObject` argument, but was afraid of causing too much ripple in the `ValidateUpdate` interface if it was modified on the validation ratcheting codepath to take a `CorrelatedObject` instead of `old` and `new` separately. 

Very open to suggestions

Reply:
Can we add a `ValidateUpdateOptions` struct to ValidateUpdate, where `nil` is an allowed value (and means there are not supplemental options, and then piggy-back on that to wire this in? 

There will be a one-time cost to wire in `ValidateUpdateOptions`, which I don't love, but I think I'd prefer that using `context.Context`.

Reply:
refactored. PTAL!